lib/src/main/java/graphql/nadel/engine/blueprint/IntrospectionService.kt

```diff
@@ -25,22 +50,266 @@ fun interface NadelIntrospectionRunnerFactory {
     fun make(schema: GraphQLSchema): ServiceExecution
 }
 
-open class NadelDefaultIntrospectionRunner(schema: GraphQLSchema) : ServiceExecution {
-    private val graphQL = GraphQL.newGraphQL(schema).build()
+typealias IntrospectionDataFetcher = GraphQLIntrospectionDataFetcher<*>
+
+open class NadelDefaultIntrospectionRunner(
+    private val schema: GraphQLSchema,
+) : ServiceExecution {
+    private val dataFetchers: Map<FieldCoordinates, IntrospectionDataFetcher> = run {
+        Introspection::class.java.getDeclaredField("introspectionDataFetchers")
+            .also {
+                it.isAccessible = true
+            }
+            .get(null)
+            .let { original ->
+                original as Map<FieldCoordinates, IntrospectionDataFetcher>
+                original.toMutableMap()
+                    .also { copy ->
+                        // todo: add other system coordinate ones
+                        copy[makeFieldCoordinates(schema.queryType.name, "__schema")] =
+                            Introspection.SchemaMetaFieldDefDataFetcher
+                        copy[makeFieldCoordinates(schema.queryType.name, "__type")] =
+                            Introspection.TypeMetaFieldDefDataFetcher
+                        copy[makeFieldCoordinates(schema.queryType.name, "__typename")] =
+                            GraphQLIntrospectionDataFetcher<Any> {
+                                schema.queryType.name
+                            }
+
+                        also {
+                            val coords = makeFieldCoordinates("__Type", "kind")
+                            val defaultImpl = copy[coords]
+                                ?: throw IllegalStateException("No __Type.kind data fetcher")
+                            copy[coords] =
+                                GraphQLIntrospectionDataFetcher<Any> { env ->
+                                    defaultImpl.get(env)?.toString()
```

Comment:
Why do you need to `.toString` the original implementation value?

The original returns

```
public enum TypeKind {
        SCALAR,
        OBJECT,
        INTERFACE,
        UNION,
        ENUM,
        INPUT_OBJECT,
        LIST,
        NON_NULL
    }
```    
and an enum can `.toString` no?`

Reply:
Hmmm looking at your benachmarks more close - we have

134.819 ms/op --> 30.656 ms /op

which is a fantastic saving and more than I expected to see from avoiding parse/validate / execute

So where is the savings coming from?  is steping through a ENF THAT much faster

Youe still doing the same amount of object allocations of DFEs etc and the same amount of propery fetching no?

Reply:
The enum type is fine because Jackson will eventually map it to a `String` anyway, but for the test result I'm comparing

`TypeKind.SCALAR` to `"SCALAR"`

Which was raising errors